src/core/dbus.c

```diff
@@ -977,7 +977,7 @@ static int bus_init_private(Manager *m) {
                 char *p = sa.un.sun_path;
                 const char *e;
 
-                e = secure_getenv("XDG_RUNTIME_DIR");
+                e = getenv("XDG_RUNTIME_DIR");
```

Comment:
The difference between `getenv()` and `secure_getenv()` is that if there is the later will return NULL if the user is gaining privileges (via setuid or file capabilities).

My rational here (and I might be wrong again) is that if we control the fact that the binary is not setuid (well that patch still needs to be proposed) and has no file capabilities set (or at least we drop them at the beginning of the execution like with the previous patch of the pull request) there is no need of using the secure variant of `getenv()`. Again I might be wrong here.

sd-bus (and libraries) in general are different as they cannot start dropping capabilities or exit(1) if the executable has file capabilities or setuid bits set. But then of course this might lead to some code duplication